pkg/controller/atlasdeployment/customzonemapping_test.go

```diff
@@ -0,0 +1,148 @@
+package atlasdeployment
+
+import (
+	"testing"
+
+	v1 "github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1"
+)
+
+type CMZTestData struct {
+	desired             []v1.CustomZoneMapping
+	existing            map[string]string
+	czmMap              map[string]string
+	expectedToCreate    bool
+	expectedToBeDeleted bool
+}
+
+const (
+	zone1     = "Zone 1"
+	zone2     = "Zone 2"
+	location1 = "CA"
+	location2 = "US"
+)
+
+func runCMZTest(t *testing.T, testData *CMZTestData) {
+	shouldCreate, shouldDelete := compareZoneMappingStates(testData.existing, testData.desired, testData.czmMap)
+	if shouldCreate != testData.expectedToCreate {
+		t.Errorf("expected to shouldCreate %v, got %v", testData.expectedToCreate, shouldCreate)
+	}
+	if shouldDelete != testData.expectedToBeDeleted {
+		t.Errorf("expected to shouldDelete %v, got %v", testData.expectedToBeDeleted, shouldDelete)
+	}
+}
+
+func TestCompareZoneMappingStates_ShouldDoNothing(t *testing.T) {
+	data := &CMZTestData{
+		desired: []v1.CustomZoneMapping{
+			{
+				Zone:     zone1,
+				Location: location1,
+			},
+			{
+				Zone:     zone2,
+				Location: location2,
+			},
+		},
+		existing: map[string]string{
+			location1: "1",
+			location2: "2",
+		},
+		czmMap: map[string]string{
+			"1": zone1,
+			"2": zone2,
+		},
+		expectedToCreate:    false,
+		expectedToBeDeleted: false,
+	}
+	runCMZTest(t, data)
+}
```

Comment:
Not an error, but consider to using table driven tests like https://github.com/mongodb/mongodb-atlas-kubernetes/blob/937fa344d4f13a24a6d491ce5aeaf84e89c88442/pkg/controller/customresource/customresource_test.go#L85-L91

Reply:
Done